### ISSUE TITLE
1.6.8x cannot handle an empty fpcsrc folder

### ISSUE THREAD

**Issue by gcarreno** [LongDirtyAnimAlf/fpcupdeluxe#258]
Hey Don,

As mentioned on #257 I'm posting this as an issue. If you think it's to small a corner case you won't fix it, I'm quite fine with it.

So, as the title says, if you have an empty `fpcsrc`, `fpcupdeluxe` will error out. Well, I say error out, since it's unable to download the sources and carry on, but to be more precise, when it _errors out_ what it shows in the log as the last line is: `fpcupdeluxe success` which is even more baffling :)

So I'll leave this one for you to try and reproduce and then try and patch.

My humble suggestion is to list `fpcsrc` folder contents and if it's 0, then delete it and carry on. In cases where it's > 0, I'm guessing `SVN` will take care of it. OR... at least make sure there is a `.svn` folder and then carry on? Don't know, ball is in your court.

Cheers,
Gus

**Comment by LongDirtyAnimAlf** [comment]
And empty fpcsrc should never happen And should never cause a problem. This has never been reported before.
I could try to reproduce, if you can reproduce it yourself and show me the steps to follow.
Perhaps something is wrong with fpcupdeluxe.

**Comment by gcarreno** [comment]
Hi Don,

I was the cause of an empty `fpcsrc`, not `fpcupdeluxe`.

Remember I said, on the issue with the failing version detection (#257 I think), that I deleted everything inside `fpcsrc` to make `fpcupdeluxe` repopulate it and see if the version thing went away?

So, in terms of reproduction, it's quite easy, just empty the folder, manually, then run `fpcupdeluxe`. I forget if I had `fpcupdeluxe` running when I emptied `fpcsrc`. But it's quite easy to test for both.

If this is a condition you're not willing to test, I think I've mentioned it also, I won't complain ;)

Cheers,
Gus
